crawler/urlJobProcessor.js

```diff
@@ -1,36 +1,106 @@
 import { default as fetch } from 'node-fetch';
+import { default as cheerio } from 'cheerio';
+import { default as urlModule } from 'url';
 
 const probes = [];
 
-function configure(/* opts */) {
-  // TODO: Set up probes. They'll be specified somehow in opts and we
-  // should be able to load each probe individually. Maybe something
-  // like `require('./probe-' + probeName + '.js');`
+function configure(opts) {
+  if (!opts.probes || !opts.probes.length) {
+    return Promise.reject(new Error('At least one urlJobProcessor probe must be specified'));
+  }
+
+  const ret = [Promise.resolve()];
+
+  opts.probes.forEach((probeOpts) => {
+    if (!probeOpts.name) {
+      return Promise.reject(new Error('Probe without name encountered'));
+    }
+
+    // We allow the probe to be specified by name or by path. If specified by name,
+    // we expect it to be a known subdir.
+    const path = probeOpts.path || `./urlJobProcessorProbes/${probeOpts.name}.js`;
+    const probeModule = require(path).default;
+
+    if (!probeModule.runProbe) {
+      return Promise.reject(new Error(`Probe ${probeOpts.name} does not implement runProbe`));
+    }
+
+    let p = Promise.resolve();
+    if (probeModule.configure) {
+      p = probeModule.configure(probeOpts.opts || {});
+    }
+
+    ret.push(p.then(() => {
+      probes.push({
+        name: probeOpts.name,
+        probeFn: probeModule.runProbe,
+      });
```

Comment:
> Do probes now fail when they are misconfigured or also when they simply could not find something they checked for?

It sounds like we're talking about three separate concepts here. One is "probe misconfiguration" where the `configure` function of a probe throws or rejects. Another is "probe failure" where the `runProbe` function of a probe throws or rejects. The last is "negative probe result" where the probe just doesn't find what it is looking for. This line note is in the `configure` function so I'll address only the first concept here.

> If one probe failing makes the whole test fail we get into a loop where a whole crawl can loop into the re-test queue because it continuously fails. The re-test queue should be just used for sites that we could not reach at the time, I thought.

A probe failing the `configure` step will cause `urlJobProcessor.configure` to reject which will stop the crawl before it even begins. This is necessary: We don't want to start a crawl where we're going to be getting bogus data. We want to fix the configuration issue before starting a crawl.